### PR TITLE
MM-44735_Freemium: Restrict create team from mobile view menu

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
When in freemium, teams are limited to just one. We need to support disabling the ability to create a team if they have reached the limit:
- Mobile view (<769 px width)
- Directly accessing URL `/create_team`

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44735
https://mattermost.atlassian.net/browse/MM-44733

##### Mobile view menu
Screen Shot 2022-06-16 at 11 36 15 AM

##### URL `/create_team`
Screen Shot 2022-06-17 at 11 20 09 AM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -2992,6 +2992,8 @@
   "create_post.shortcutsNotSupported": "Keyboard shortcuts are not supported on your device.",
   "create_post.write": "Write to {channelDisplayName}",
   "create_team.agreement": "By proceeding to create your account and use {siteName}, you agree to our [Terms of Use]({TermsOfServiceLink}) and [Privacy Policy]({PrivacyPolicyLink}). If you do not agree, you cannot use {siteName}.",
+  "create_team.createTeamRestricted.message": "This is a paid feature, available with a free 30-day trial. Please contact your System Administrator to resolve this.",
+  "create_team.createTeamRestricted.title": "Professional feature",
```
Comment (julmondragon) [inline i18n/en.json:2996]: Used these title/message to speed up development. @annelieseklein please let us know any updated needed.

File: components/select_team/index.ts [hunk 2]
```diff
@@ -47,6 +50,7 @@ function mapStateToProps(state: GlobalState) {
         siteURL: config.SiteURL,
         totalTeamsCount: state.entities.teams.totalCount || 0,
         isCloud,
+        isFreeTrial,
```
Comment (julmondragon) [inline components/select_team/index.ts:53]: This is for consistence across restricted features, not using limits during trial.

File: components/create_team/create_team.tsx [hunk 3]
```diff
@@ -71,8 +76,18 @@ export default class CreateTeam extends React.PureComponent<Props & RouteCompone
...
+            isFreeTrial,
+            usageDeltas: {
+                teams: {
+                    active: usageDeltaTeams,
+                },
+            },
         } = this.props;
 
+        const teamsLimitReached = usageDeltaTeams >= 0;
+        const createTeamRestricted = isCloud && !isFreeTrial && teamsLimitReached;
```
Comment (marianunez) [inline components/create_team/create_team.tsx:89]: If I'm not mistaken, the limits are updated on free trial so we shouldn't need to check for free trial given that the limit will not be there

File: i18n/en.json (hunk 1 above)
Comment (marianunez) [reply i18n/en.json:2996]: It seems that this text is shown but doesn't show what feature its hiding. Should we include something in the text that create teams have reached their limit on their current plan or something that is clear what we are referring as a Professional feature. 

@johndavidlugtu any suggestions? (See screenshot in the PR description)

File: components/select_team/index.ts (hunk 2 above)
Comment (marianunez) [reply components/select_team/index.ts:53]: Not sure I understand what you mean here - during trial they have no limits so checking the limits at all times should be enough

File: components/create_team/create_team.tsx [hunk 4]
```diff
@@ -92,29 +107,47 @@ export default class CreateTeam extends React.PureComponent<Props & RouteCompone
...
-                                    render={(props) => (
-                                        <TeamUrl
-                                            state={this.state}
-                                            updateParent={this.updateParent}
-                                            {...props}
+                                    </h5>
+                                    <div>
+                                        <FormattedMessage
+                                            id='create_team.createTeamRestricted.message'
+                                            defaultMessage='This is a paid feature, available with a free 30-day trial. Please contact your System Administrator to resolve this.'
```
Comment (marianunez) [inline components/create_team/create_team.tsx:122]: Can we add some tests for this new case?

**Comment by johndavidlugtu** [comment]
>It seems that this text is shown but doesn't show what feature its hiding. Should we include something in the text that create teams have reached their limit on their current plan or something that is clear what we are referring as a Professional feature.

@marianunez is this something we're displaying for end users _and_ admins?

**Comment by marianunez** [comment]
> @marianunez is this something we're displaying for end users _and_ admins?

This will be displayed for any user that has create team permission but can't because the workspace reached it's limit on teams

**Comment by johndavidlugtu** [comment]
**Professional feature**

Your workspace plan has reached the limit on the number of teams. Create unlimited teams with a free 30-day trial. Contact your System Administrator.

@marianunez what do you think?

File: components/select_team/index.ts (hunk 2 above)
Comment (julmondragon) [reply components/select_team/index.ts:53]: Limits are still hanging when trial is started, this just skips the flakiness in the service data for this use case.

File: components/create_team/create_team.tsx (hunk 3 above)
Comment (julmondragon) [reply components/create_team/create_team.tsx:89]: Same as above comment.

**Comment by julmondragon** [comment]
> 

Updated @johndavidlugtu @marianunez

**Review by stevemudie — APPROVED** [review]
Looks good in mobile view @julmondragon!

**Review by marianunez — APPROVED** [review]
Thanks @julmondragon!

**Review by pvev — APPROVED** [review]
LGTM! nice work @julmondragon

**Pull request merged**